### PR TITLE
feat: mock chat model

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Mock Chat Model that can be used for testing, either when building the app or for automated tests.

### Test plan

Added test for Application using the new mock chat model

### CODE REVIEW

File: packages/core/src/__tests__/application.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { expect, test, vitest } from 'vitest';
+import { createApp } from '../application.js';
+import { Message } from '../domain.js';
+import { createMockModel } from '../mock/mock-model.js';
+
+const messages: Message[] = [{ role: 'user', content: 'Hello' }];
+
+test('basic non-streaming test', async () => {
+  const testModel = createMockModel({ delay: 0, seed: 3 });
```
Comment (Q1w1N) [inline packages/core/src/__tests__/application.test.ts:9]: Maybe add one small test for delay as well?

File: packages/core/src/mock/mock-model.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  'Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+  'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.',
+  'Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt.',
+  'Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem.',
+  'Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?',
+  'Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?',
+];
+
+export type MockModelConfig = {
+  items?: string[];
```
Comment (Q1w1N) [inline packages/core/src/mock/mock-model.ts:17]: mockResponses or any other name that would suggest that this is a "container" for that.

File: packages/core/src/mock/mock-model.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      if (context.onPartialResponse) {
+        let accumulator = '';
+        for (const token of tokens) {
+          await sleep(delay);
+          accumulator += token;
+          context.onPartialResponse(accumulator);
+        }
+      }
+
+      await sleep(delay);
```
Comment (Q1w1N) [inline packages/core/src/mock/mock-model.ts:43]: Won't putting sleep for 0 delay cause any issues with event loop and micro/macro tasks queue?

**Review by Q1w1N — APPROVED** [review]
Minor questions/suggestions. Beside that looks awesome

File: packages/core/src/mock/mock-model.ts (hunk 3 above)
Comment (mdjastrzebski) [reply packages/core/src/mock/mock-model.ts:43]: Nah, 0 delay will just pause till next macro task

File: packages/core/src/mock/mock-model.ts (hunk 2 above)
Comment (mdjastrzebski) [reply packages/core/src/mock/mock-model.ts]: changed to `responses`

File: packages/core/src/__tests__/application.test.ts (hunk 1 above)
Comment (mdjastrzebski) [reply packages/core/src/__tests__/application.test.ts]: Not worth it, delay is for normal testing. For tests it will only slow it down.

**Pull request merged**